pkg/telemetry/metrics_helper.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package telemetry
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/cli"
+	configlib "github.com/vmware-tanzu/tanzu-plugin-runtime/config"
+	configtypes "github.com/vmware-tanzu/tanzu-plugin-runtime/config/types"
+)
+
+func getEndpointSHA(plugin *cli.PluginInfo) string {
+	cfg, err := configlib.GetClientConfig()
+	if err != nil {
+		return ""
+	}
+	curCtxMap, err := cfg.GetAllCurrentContextsMap()
+	if err != nil || curCtxMap == nil {
+		return ""
+	}
+
+	if plugin.Target == configtypes.TargetK8s {
+		return getCurrentK8SContextSHA(curCtxMap)
+	} else if plugin.Target == configtypes.TargetTMC {
+		return getCurrentTMCContextSHA(curCtxMap)
+	}
```

Comment:
Can we make this generic and not have separate functions like this for each active context for the target?

In the future, we would like to decouple the context/target, and having a specific function for each target does not make sense considering that.

What is the reason for doing calculating SHA256 for both the `Context` objects differently?

Reply:
> Can we make this generic and not have separate functions like this for each active context for the target?
> 
> In the future, we would like to decouple the context/target, and having a specific function for each target does not make sense considering that.

> What is the reason for doing calculating SHA256 for both the Context objects differently?

If we take SHA256 for the contexts same (taking SHA256 of Context structure) , it would result in a new SHA everytime the `IDToken` or `accessToken` changes and these are shortlived. For Kubernetes context, we can take the SHA of the complete context as there are no shortlived values there.

In future if we decouple the context/target, we have to find out other means to detect the context that plugin uses and hash the right context. So it would be incremental change

Reply:
@anujc25 The goal of the endpoint value computation is to produce the most stable, non-PII value that can be used to accurately distinguish among different endpoints the commands are targeting. The heuristics used here, which relies on using the context data,  needs to be tuned differently for different styles of context for best effect.

@prkalle  That said, perhaps the fn names (getCurrentK8SContextSHA. ...) is setting off a wrong impression that we have multiple ways of computing a SHA.  Maye names like getEndpointSHAForK8SContext, getEndpointSHAForGlobalContext would be more reflective of what the function does.

Alternatively, (similar to what @anujc25 suggested) can we just write a single function computeEndpointSHAForContext that does the right thing depending on what is set in the context? (Some outer code will have to determine that context to compute and pass that into this function)

Reply:
Updated the PR with single function `computeEndpointSHAForContext`that computes the endpoint SHA for the context,